src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -430,13 +430,13 @@
     <value>Invalid leading zero before '{0}'.</value>
   </data>
   <data name="MetadataCannotParsePreservedObjectToImmutable" xml:space="preserve">
-    <value>Cannot parse a JSON object containing metadata properties like '$id' into an array or immutable collection type. Type '{0}'.</value>
+    <value>Cannot parse a JSON object containing metadata properties like '$id' or '$type' into an array or immutable collection type. Type '{0}'.</value>
```

Comment:
should this use argument considering it's configurable?

Reply:
I thought about it, but ultimately decided against surfacing the actual custom property name. Ultimately the message is meant to provide examples, not an exhaustive list of cases.